icu4c/source/test/intltest/numbertest_permutation.cpp

```diff
@@ -0,0 +1,189 @@
+// © 2017 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include <vector>
+#include <iostream>
+#include <fstream>
+
+#include "numbertest.h"
+#include "ucbuf.h"
+#include "unicode/numberformatter.h"
+
+void NumberPermutationTest::runIndexedTest(int32_t index, UBool exec, const char*& name, char*) {
+  if (exec) {
+    logln("TestSuite NumberPermutationTest: ");
+  }
+  TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(testPermutations);
+  TESTCASE_AUTO_END;
+}
+
+static const char16_t* kSkeletonParts[] = {
+    // Notation
+    u"compact-short",
+    u"scientific/+ee/sign-always",
+    0,
+    // Unit
+    u"percent",
+    u"currency/EUR",
+    u"measure-unit/length-furlong",
+    0,
+    // Unit Width
+    u"unit-width-narrow",
+    u"unit-width-full-name",
+    0,
+    // Precision
+    u"precision-integer",
+    u".000",
+    u".##/@@@+",
+    u"@@",
+    0,
+    // Rounding Mode
+    u"rounding-mode-floor",
+    0,
+    // Integer Width
+    u"integer-width/##00",
+    0,
+    // Scale
+    u"scale/0.5",
+    0,
+    // Grouping
+    u"group-on-aligned",
+    0,
+    // Symbols
+    u"latin",
+    0,
+    // Sign Display
+    u"sign-accounting-except-zero",
+    0,
+    // Decimal Separator Display
+    u"decimal-always",
+    0,
+};
+
+static const double kNumbersToTest[] {
+    0,
+    91827.3645,
+    0.22222
+};
```

Comment:
Optional: add a negative value? Or make one of these negative?

Reply:
Done